cni-plugin/Dockerfile

```diff
@@ -0,0 +1,27 @@
+## compile cni-plugin utility
+FROM gcr.io/linkerd-io/go-deps:7db9daa2 as golang
+COPY ./proxy-init/cmd /go/src/github.com/linkerd/linkerd2/proxy-init/cmd
+COPY ./proxy-init/iptables /go/src/github.com/linkerd/linkerd2/proxy-init/iptables
+COPY ./pkg/k8s /go/src/github.com/linkerd/linkerd2/pkg/k8s
+COPY ./pkg/version /go/src/github.com/linkerd/linkerd2/pkg/version
+COPY ./controller/gen/public /go/src/github.com/linkerd/linkerd2/controller/gen/public
+COPY ./controller/gen/common/healthcheck /go/src/github.com/linkerd/linkerd2/controller/gen/common/healthcheck
+WORKDIR /go/src/github.com/linkerd/linkerd2
+COPY ./cni-plugin ./cni-plugin
+RUN CGO_ENABLED=0 GOOS=linux go build -o /go/bin/linkerd-cni -v ./cni-plugin/
+
+FROM alpine:3.8
```

Comment:
Looks like the `gcr.io/linkerd-io/base:2017-10-30.01` doesn't have a version of `jq` that supports the `--argjson` flag. That came in version 1.5 of `jq`

Reply:
I tried with the `jq` flag `--arg` and because the value coming in is json, `jq` escapes the quotes and we end up with a conf file that looks like: 

```
{
  "name": "k8s-pod-network",
  "cniVersion": "0.3.0",
  "plugins": [
    {
      "type": "calico",
      "log_level": "info",
      "etcd_endpoints": "http://10.96.232.136:6666",
      "etcd_key_file": "",
      "etcd_cert_file": "",
      "etcd_ca_cert_file": "",
      "mtu": 1440,
      "ipam": {
        "type": "calico-ipam"
      },
      "policy": {
        "type": "k8s"
      },
      "kubernetes": {
        "kubeconfig": "/etc/cni/net.d/calico-kubeconfig"
      }
    },
    {
      "type": "portmap",
      "snat": true,
      "capabilities": {
        "portMappings": true
      }
    },
    "{
  \"name\": \"linkerd-cni\",
  \"type\": \"linkerd-cni\",
  \"log_level\": \"debug\",
  \"policy\": {
      \"type\": \"k8s\",
      \"k8s_api_root\": \"https://10.96.0.1:443\",
      \"k8s_auth_token\": \"eyJhbGciOiJSUzI1NiIsImtpZCI6IiJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJsaW5rZXJkIiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZWNyZXQubmFtZSI6ImxpbmtlcmQtY25pLXRva2VuLXZrNXA2Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZXJ2aWNlLWFjY291bnQubmFtZSI6ImxpbmtlcmQtY25pIiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZXJ2aWNlLWFjY291bnQudWlkIjoiYWIzOGFjZTYtMTlkNC0xMWU5LTgwNTAtMDgwMDI3MGQxNDY4Iiwic3ViIjoic3lzdGVtOnNlcnZpY2VhY2NvdW50OmxpbmtlcmQ6bGlua2VyZC1jbmkifQ.K70O67wSiVH_KlixDaCGWLCg-Qcqpl45Mc1pjBCg9zE8RaTPnKUwfVoy6lHITzey6mJqAevDJaZg3b4ke0imUH1jRhzJ8j-nyj40IrwCQVLsr3yTtGyrvzySsOZXSsf9J5WqReE5-JTeWSMd9eiNTO8ah8CxpmB8hlnXRWtDUUUocgftjfxAtsPOzxGPEKlNE9m6AVoMbx7bfXWjmXYC2vszZoLHQ8DgZhvZRluiDJEdmeSPZDcUGqVi1oaCNHA4jvYKz6RNsrLjoKXlWwHgpAbwJIfr1DFXPoxx0yVAXBYi0RjRzh4k4Em9vy3Y7HDU9hKL-Me1KDh7RLE2ZZoRiQ\"
  },
  \"kubernetes\": {
      \"kubeconfig\": \"/etc/cni/net.d/ZZZ-linkerd-cni-kubeconfig\"
  },
  \"linkerd\": {
    \"incoming-proxy-port\": 4143,
    \"outgoing-proxy-port\": 4140,
    \"proxy-uid\": 2102,
    \"ports-to-redirect\": [],
    \"inbound-ports-to-ignore\": [4190,4191],
    \"outbound-ports-to-ignore\": [],
    \"simulate\": false
  }
}"
  ]
}
```

which results in `Error loading CNI config list file /etc/cni/net.d/10-calico.conflist: error parsing configuration list: invalid character '\n' in string literal`

Reply:
Ah shucks -- sorry about that. I didn't realize `jq` is out of date. In that case I'd prefer to still use the official base image, but upgrade `jq` manually as part of the image build. Something like:

```
FROM gcr.io/linkerd-io/base:2017-10-30.01
WORKDIR /linkerd

RUN curl -kL -o $(which jq) https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64
```

(looks like we need the `-k` cause the base image doesn't include ca certificates -- we should fix that separately)